keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -126,16 +127,45 @@ class TitusClusterHandler(
             else -> diff.upsertServerGroupJob() + resource.spec.deployWith.toOrcaJobProperties()
           }
 
+          var tags: Set<String> = emptySet()
+          val version = when {
+            diff.isCapacityOnly() -> null
+            else -> {
+              // calculate the version for the digest
+              tags = getTagsForDigest(desired.container, desired.location.account).filterNot { it == "latest" }.toSet()
+              if (tags.size == 1) {
+                tags.first()
+              } else {
+                log.debug("Container digest ${desired.container} has multiple tags: $tags")
```

Comment:
Would it be safe to sort the tags in descending order and pick the first? As in, the latest tag applied?

Reply:
On second thought, never mind. I'm assuming the UI will have a link that takes the user to the image, where they'll see it has multiple tags. Perhaps just use the short hash for the digest instead?

Reply:
👍 I'll shorten the hash here so it's not so long. Until we see a bunch of situations where this happens i'm not sure what the right behavior is - but I think you're right, you'll just need to look at the image yourself.